source/common/config/utility.cc

```diff
@@ -44,10 +44,7 @@ void Utility::translateApiConfigSource(
     envoy::config::core::v3::GrpcService* grpc_service = api_config_source.add_grpc_services();
     grpc_service->mutable_envoy_grpc()->set_cluster_name(cluster);
   } else {
-    if (api_type == ApiType::get().UnsupportedRestLegacy) {
-      api_config_source.set_api_type(envoy::config::core::v3::ApiConfigSource::
-                                         hidden_envoy_deprecated_UNSUPPORTED_REST_LEGACY);
-    } else if (api_type == ApiType::get().Rest) {
+    if (api_type == ApiType::get().Rest) {
```

Comment:
I think this should be just:
```
} else {
  ASSERT(api_type == ApiType::get().Rest);
  api_config_source.set_api_type(envoy::config::core::v3::ApiConfigSource::REST);
...
```
You should also remove `UnsupportedRestLegacy` and any references to this in the code in this PR.

Reply:
Done. Thnaks for review